pkg/kv/postgres/store.go

```diff
@@ -288,20 +294,14 @@ func (s *Store) scanInternal(ctx context.Context, partitionKey []byte, options k
 		err  error
 	)
 
-	// limit set to the minimum between option's limit and the configured scan page size
-	limit := s.Params.ScanPageSize
-	if options.BatchSize > 0 && options.BatchSize < limit {
-		limit = options.BatchSize
-	}
-
 	if options.KeyStart == nil {
-		rows, err = s.Pool.Query(ctx, `SELECT partition_key,key,value FROM `+s.Params.SanitizedTableName+` WHERE partition_key=$1 ORDER BY key LIMIT $2`, partitionKey, limit)
+		rows, err = s.Pool.Query(ctx, `SELECT partition_key,key,value FROM `+s.Params.SanitizedTableName+` WHERE partition_key=$1 ORDER BY key LIMIT $2`, partitionKey, options.BatchSize)
```

Comment:
I'd consider returning to limit, IMO it's confusing that options.BatchSize was edited